### DISCUSSION TITLE
Decomposition of 1026 to 1025 + 102E

### DISCUSSION THREAD

**Discussion by ohbendy** [harfbuzz/harfbuzz#5030] [General] [answered]
Following a bit of discussion on BlueSky, we wanted to raise the possibility of overriding this canonical decomposition.

In Myanmar script, u+1026 normally looks identical to 1025 + 102E, and indeed Unicode prescribes a canonical decomposition 1026 -> 1025 102E. 

![Burmese U UU](https://github.com/user-attachments/assets/9ac0f669-5365-4b65-aacc-b5272fad3e06)

This is probably ok for general Burmese text, though see my next comment showing independent UU in a different form in a Sanskrit text.

However. For Arakanese, there are plenty of examples where 1026 takes a different form from 1025 102E. Here are some manuscript examples of 1026, only the third version is 1025 + 102E:

![Versions of Arakanese UU](https://github.com/user-attachments/assets/27e8d5e4-60b0-44c4-a588-2f84002b9540)

Ideally these would be stored as 1026, 1025 102D, 1025 102E, 1025 102F. But if 1026 is unavoidably decomposed to 1025 102E, there's no way to obtain the first glyph for 1026. 

Currently I'm recomposing 1025 102E -> 1026, but this means we can't also have the actual 1025 102E (the third manuscript image).

![Arakanese U UU](https://github.com/user-attachments/assets/6f35d6ad-936a-4292-be67-e56cebaa7c6d)

Perhaps all this is an argument for encoding at least this one Arakanese character distinct from the Burmese , but perhaps it could be a HB override?

**Comment by behdad** [comment]
cc @dscorbett @jfkthame @PeterCon @nedley

**Comment by ohbendy** [comment]
As a related point, independent UU also takes a form different from U + dependent ii in some Burmese manuscripts. This is from a Sanskrit language manuscript where they're listing the independent vowels one by one. We can see အ အာ ဣ ဤ ဥ ဦ with UU written as ဥဲ here. Arguably that can be stored as 1025 1032 but I'd consider that a workaround as linguistically it should be 1026 because it's really an atomic character.

![Burmese Independent Vowels](https://github.com/user-attachments/assets/c042441a-f81c-4d9b-ba45-f433f1f5ecf7)

**Comment by PeterConstable** [comment, marked as answer]
You're proposing something that would not conform to Unicode:

>C6A process shall not assume that the interpretations of two canonical-equivalent character sequences are distinct.

(https://unicode-core-spec.pages.dev/chapter-3/#G33436)

I think you should bring an issue to the Unicode Technical Committee / Script Encoding Working Group to discuss alternative solutions for Arakanese.

@miloush @roozbehp

**Comment by roozbehp** [comment]
I agree with @PeterConstable here. That would not be conforming.

As for the actual issue at hand, in the four-examples image, the first example probably shouldn't be represented using U+1026, but either a new character or a sequence of characters (perhaps <U+1025, U+102F>?).

As @PeterConstable said, instead of a non-complying hack in HarfBuzz, you may need a solution in proper Unicode. Write a document explaining the issues, and Unicode Script Encoding Working Group can help progress it towards a solution.
